account_payment_partner/models/account_move.py

```diff
@@ -0,0 +1,155 @@
+                move.payment_mode_filter_type_domain = "outbound"
+            else:
+                move.payment_mode_filter_type_domain = False
+
+    @api.depends("partner_id", "move_type")
+    def _compute_partner_bank_filter_type_domain(self):
+        for move in self:
+            if move.move_type in ("out_invoice", "in_refund"):
+                move.partner_bank_filter_type_domain = move.bank_partner_id
+            elif move.move_type in ("in_invoice", "out_refund"):
+                move.partner_bank_filter_type_domain = move.commercial_partner_id
+            else:
+                move.partner_bank_filter_type_domain = False
+
+    @api.depends("partner_id", "company_id")
+    def _compute_payment_mode(self):
+        for move in self:
+            move.payment_mode_id = move.payment_mode_id
+            if move.company_id and move.payment_mode_id.company_id != move.company_id:
+                move.payment_mode_id = False
+            if move.partner_id:
+                partner = move.with_company(move.company_id.id).partner_id
+                if move.move_type == "in_invoice":
+                    move.payment_mode_id = partner.supplier_payment_mode_id
+                elif move.move_type == "out_invoice":
+                    move.payment_mode_id = partner.customer_payment_mode_id
+                elif (
+                    move.move_type in ["out_refund", "in_refund"]
+                    and move.reversed_entry_id
+                ):
+                    move.payment_mode_id = (
+                        move.reversed_entry_id.payment_mode_id.refund_payment_mode_id
+                    )
+                elif not move.reversed_entry_id:
+                    if move.move_type == "out_refund":
+                        move.payment_mode_id = (
+                            partner.customer_payment_mode_id.refund_payment_mode_id
+                        )
+                    elif move.move_type == "in_refund":
+                        move.payment_mode_id = (
+                            partner.supplier_payment_mode_id.refund_payment_mode_id
+                        )
+
+    @api.onchange("partner_id")
+    def _onchange_partner_id(self):
+        """Force compute because the onchange chain doesn't call
+        ``_compute_partner_bank``.
+        """
+        res = super()._onchange_partner_id()
+        self._compute_partner_bank()
+        return res
+
+    @api.depends("partner_id", "payment_mode_id")
+    def _compute_partner_bank(self):
+        for move in self:
+            # No bank account assignation is done for out_invoice as this is only
+            # needed for printing purposes and it can conflict with
+            # SEPA direct debit payments. Current report prints it.
+            def get_bank_id():
+                return move.commercial_partner_id.bank_ids.filtered(
+                    lambda b: b.company_id == move.company_id or not b.company_id
+                )[:1]
```

Comment:
```suggestion
                return fields.first(move.commercial_partner_id.bank_ids.filtered(
                    lambda b: b.company_id == move.company_id or not b.company_id
                ))
```

Reply:
@i-vyshnevska Thanks for pointing out. I didn't know this method! :+1: